deltacat/compute/compactor/steps/materialize.py

```diff
@@ -181,27 +175,52 @@ def _materialize(
             [sc._DEDUPE_TASK_IDX_COLUMN_NAME]
         )
 
-        # Write manifests up to max_records_per_output_file
-        # TODO(raghumdani): Write exactly the same number of records into each file to
-        # produce a read-optimized view of the tables.
-        if compacted_tables and \
-                total_record_count + record_count > max_records_per_output_file:
-            materialized_results.append(_materialize(compacted_tables, total_record_count))
+        if remainder_tables:
+            prev_remainder_record_count = sum([len(tbl) for tbl in remainder_tables])
```

Comment:
This may become inefficient if we end up iterating through all the remainder tables especially if there are lot of small tables and our max_records_per_output_file is too huge. Basically, increasing the complexity from O(N) to skim through this to O(N^2).